Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -86,21 +86,25 @@ unsigned ScriptDisallowedScope::s_count = 0;
 ScriptDisallowedScope::EventAllowedScope* ScriptDisallowedScope::EventAllowedScope::s_currentScope = nullptr;
 #endif
 
-ALWAYS_INLINE auto ContainerNode::removeAllChildrenWithScriptAssertion(ChildChange::Source source, NodeVector& children, DeferChildrenChanged deferChildrenChanged) -> DidRemoveElements
+ALWAYS_INLINE auto ContainerNode::removeAllChildrenWithScriptAssertion(ChildChange::Source source, NodeVector& children, DeferChildrenChanged deferChildrenChanged) -> RemoveAllChildrenResult
 {
     ASSERT(children.isEmpty());
     collectChildNodes(*this, children);
 
+    auto childrenDeletionCanBeDelayed = ChildrenDeletionCanBeDelayed::Yes;
+
     if (UNLIKELY(isDocumentFragmentForInnerOuterHTML())) {
         ScriptDisallowedScope::InMainThread scriptDisallowedScope;
         RELEASE_ASSERT(!connectedSubframeCount() && !hasRareData() && !wrapper());
         bool hadElementChild = false;
+        int treeSize = 0;
         while (RefPtr child = m_firstChild.get()) {
+            treeSize++;
             hadElementChild |= is<Element>(*child);
             removeBetween(nullptr, child->protectedNextSibling().get(), *child);
         }
         document().incDOMTreeVersion();
-        return hadElementChild ? DidRemoveElements::Yes : DidRemoveElements::No;
+        return { treeSize, hadElementChild ? DidRemoveElements::Yes : DidRemoveElements::No, childrenDeletionCanBeDelayed };
```

Comment:
We didn't check whether each node can be deleted asynchronously or not either.
Maybe we need a third state for this enum class like "Unknown".